app/validators/types_de_champ/condition_validator.rb

```diff
@@ -1,17 +1,21 @@
 class TypesDeChamp::ConditionValidator < ActiveModel::EachValidator
-  def validate_each(procedure, attribute, types_de_champ)
-    return if types_de_champ.empty?
+  # condition are valid when
+  #   tdc.condition.left is present in upper tdcs
+  #   in case of types_de_champ_private, we should include types_de_champ_private too
+  def validate_each(procedure, attribute, scoped_types_de_champ)
+    return if scoped_types_de_champ.empty?
 
-    tdcs = if attribute == :draft_types_de_champ_private
-      procedure.draft_revision.types_de_champ_for
-    else
-      procedure.draft_revision.types_de_champ_for(scope: :public)
-    end
-
-    tdcs.each_with_index do |tdc, i|
+    scoped_types_de_champ = tdcs_with_children(procedure, scoped_types_de_champ)
```

Comment:
```suggestion
    tdcs = tdcs_with_children(procedure, scoped_types_de_champ)
```

le terme `scoped` m'embrouille un peu